app/libc/stdio.c

```diff
@@ -221,10 +186,8 @@ str_fmt(char *p, int size, int fmt)
  * SUCH DAMAGE.
  *
  */
-//#include <string.h>
-//#include <ctype.h>
-#include "c_string.h"
-#include "c_ctype.h"
+#include <string.h>
+#include <ctype.h>
 
```

Comment:
What's going on with all these repeated includes?  Was this file generated by concatenating a whole bunch of other files?

Reply:
Having nested includes is pretty standard have a look at the standard `<stdio.c>` for example.  This is why most includes use the `#ifndef _XXXXX_H_` guard around the include content.  IMO we should also just use `#pragma once` by default in our code as this speeds up compilation and is supported by the GCC toolchains.

Reply:
Yeah this was inherited. Originally they pulled in whatever bits they could from wherever they could to patch over the shonky libc.a in the SDK. With all the different copyright notices in there I didn't want to touch it more than necessary.

Reply:
OK.  I feel bad for everyone working around Espressif's... design decisions.